### PR TITLE
Bump PyTensor dependency

### PR DESCRIPTION
**Pull request by ricardoV94**

### CODE REVIEW

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
isn't this also in the precommit?

**Comment by ricardoV94** [comment]
Yeah, I'm investigating it. It has a C implementation...

**Comment by ricardoV94** [comment]
> isn't this also in the precommit?

I don't think it is anymore

**Comment by ricardoV94** [comment]
The external samplers failure is what I feared would happen with the nutpie circular dependency?

**Comment by ricardoV94** [comment]
The warning should be fixed by https://github.com/pymc-devs/pytensor/pull/225

**Comment by twiecki** [comment]
Lots of install conflicts :(.

**Comment by ricardoV94** [comment]
It's because nutpie pins the old pymc's pytensor dependency on conda. We can't test it directly as I suspected. Will remove the nutpie tests for now

File: tests/sampling/test_mcmc_external.py [hunk 1]
```diff
@@ -17,16 +17,15 @@
...
-pytest.importorskip("numpyro")
-
 # turns all warnings into errors for this module
 pytestmark = pytest.mark.filterwarnings("error")
 
 
 @pytest.mark.parametrize("nuts_sampler", ["pymc", "nutpie", "blackjax", "numpyro"])
 def test_external_nuts_sampler(recwarn, nuts_sampler):
+    if nuts_sampler != "pymc":
+        pytest.importorskip(nuts_sampler)
```
Comment (ricardoV94) [inline tests/sampling/test_mcmc_external.py:27]: I checked this works as expected and skips only the parametrization that fails to import, not all of them afterwards. Can be seen in the CI as well: https://github.com/pymc-devs/pymc/actions/runs/4282342784/jobs/7456497043#step:8:79

**Pull request merged**